packaging/os/yumrepo.py

```diff
@@ -549,40 +914,81 @@ def main():
     name = module.params['name']
     state = module.params['state']
 
-    # Rename "name" and "description" to ensure correct key sorting
-    module.params['repoid'] = module.params['name']
-    module.params['name'] = module.params['description']
-    del module.params['description']
-
-    # Instantiate the YumRepo object
-    yumrepo = YumRepo(module)
-
-    # Get repo status before change
-    yumrepo_before = yumrepo.dump()
-
-    # Perform action depending on the state
+    # Check if required parameters are present
+    if state in ['present', 'absent'] and 'name' is None:
+        module.fail_json(
+            msg="Parameter 'name' is required.")
     if state == 'present':
-        yumrepo.add()
-    elif state == 'absent':
-        yumrepo.remove()
-
-    # Get repo status after change
-    yumrepo_after = yumrepo.dump()
-
-    # Compare repo states
-    changed = yumrepo_before != yumrepo_after
-
-    # Save the file only if not in check mode and if there was a change
-    if not module.check_mode and changed:
-        yumrepo.save()
-
-    # Change file attributes if needed
-    if os.path.isfile(module.params['dest']):
-        file_args = module.load_file_common_arguments(module.params)
-        changed = module.set_fs_attributes_if_different(file_args, changed)
-
-    # Print status of the change
-    module.exit_json(changed=changed, repo=name, state=state)
+        if (
+                'baseurl' is None and
+                'mirrorlist' is None):
+            module.fail_json(
+                msg="Parameter 'baseurl' or 'mirrorlist' is required.")
+        if 'description' is None:
+            module.fail_json(
+                msg="Parameter 'description' is required.")
+
+    # Instantiate the ManagedMode object
+    mm = ManagedMode(module)
+
+    if state in ['present', 'absent']:
+        # Rename "name" and "description" to ensure correct key sorting
+        module.params['repoid'] = module.params['name']
+        module.params['name'] = module.params['description']
+        del module.params['description']
+        del module.params['ignore_repo_files']
+
+        # Define repo file name if it doesn't exist
+        if module.params['file'] is None:
+            module.params['file'] = module.params['repoid']
```

Comment:
As you said, this is obvious that this module won't search through all repo files to find if such repo is already defined. That's up to the user to deal with that and specify the `file` parameter to target the right repo file if needed.